src/browser/replay/recorder.js

```diff
@@ -68,39 +68,36 @@ export default class Recorder {
   checkoutEveryNms() {
     // Recording may be up to two checkout intervals, therefore the checkout
     // interval is set to half of the maxSeconds.
-    return (this.options.maxSeconds || 10) * 1000 / 2;
+    return ((this.options.maxSeconds || 10) * 1000) / 2;
   }
 
   /**
-   * Converts recorded events into a formatted payload ready for transport.
+   * Exports the recording span with all recorded events.
    *
    * This method takes the recorder's stored events, creates a new span with the
    * provided tracing context, attaches all events with their timestamps as span
-   * events, and then returns a payload ready for transport to the server.
+   * events, and exports the span to the tracing exporter. This is a side-effect
+   * function that doesn't return anything - the span is exported internally.
    *
    * @param {Object} tracing - The tracing system instance to create spans
-   * @param {string} replayId - Unique identifier to associate with this replay recording
-   * @returns {Object|null} A formatted payload containing spans data in OTLP format, or null if no events exist
+   * @param {Object} attributes - Attributes to add to the span
+   *  (e.g., rollbar.replay.id, rollbar.occurrence.uuid)
    */
-  dump(tracing, replayId, occurrenceUuid) {
-    const events = this._events.previous.concat(this._events.current);
+  exportRecordingSpan(tracing, attributes = {}) {
+    const events = this._collectEvents();
 
-    if (events.length < 2) {
-      logger.error('Replay recording cannot have less than 2 events');
-      return null;
+    if (events.length < 3) {
```

Comment:
+ 1 since now we have the final ending event, which is being added in `_collectEvents` now.